cpp/modmesh/inout/gmsh.cpp

```diff
@@ -94,6 +94,159 @@ void Gmsh::build_interior(const std::shared_ptr<StaticMesh> & blk)
     blk->build_boundary();
     blk->build_ghost();
 }
+
+// Check the finite state machine state transition is valid or not to check msh file format is correct
+bool Gmsh::is_valid_transition(const std::string s)
+{
+    if (last_fmt_state == FormatState::BEGIN)
+    {
+        return s == "$MeshFormat";
+    }
+    else if (last_fmt_state == FormatState::META_END || last_fmt_state == FormatState::PYHSICAL_NAME_END)
```

Comment:
It looks odd to me that you use "or" here. Why not split them into two `else if`?

Reply:
This is my bad, I have checked the [gmsh manual](https://gmsh.info/dev/doc/texinfo/gmsh.pdf) section 10.3.1.
According to the manual, after `MeshFormat`, which is the `META_END` state, it can either be followed by `$PhysicalNames` or `$Nodes`. That is why the function returns an "or" expression. However the problem is that after physical name section, it only can be followed by `Node`. Therefore this code should be changed to 
```
...
else if (last_fmt_state == FormatState::META_END)
{
    return s == "$PhysicalNames" || s == "$Nodes";
}
else if (last_fmt_state == FormatState::PYHSICAL_NAME_END)
{
    return s == "$Nodes";
}
...
```
Thanks @tigercosmos to point out this problem.
I think we can use another PR to fix this problem, it should add some test case to make sure this logic is correct!

I will file an issue to track this.